FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Api/FhirApiDataClient.cs

```diff
@@ -229,7 +229,7 @@ private async Task<string> GetResponseFromHttpRequestAsync(Uri uri, string acces
                 _logger.LogInformation(bcEx, "Broken circuit while searching from FHIR server. Reason: {0}", bcEx.Message);
 
                 throw new FhirSearchException(
-                    string.Format(Resource.FhirSearchFailed, uri),
+                    string.Format(Resource.FhirSearchFailed, uri) + $"Reason:{bcEx.Message}",
```

Comment:
Can we change FhirSearchFailed  string to accept another reason field?

Reply:
Updated it.